POEM_060.md

```diff
@@ -0,0 +1,62 @@
+POEM ID: 060  
+Title: Reports  
+authors: robfalck  
+Competing POEMs:  
+Related POEMs:  
+Associated implementation PR: N/A
+
+Status:
+
+- [x] Active
+- [ ] Requesting decision
+- [ ] Accepted
+- [ ] Rejected
+- [ ] Integrated
+
+
+## Motivation
+
+OpenMDAO is capable of outputting a variety of diagnostic information to both standard output and HTML files.
+
+The current implementation isn't completely user friendly in two ways.
+
+1. Prolific standard output can make it difficult to find relevant information when using diagnostics that print there, such as `p.list_problem_vars()`.
+2. HTML reports often require the user consult the documentation to see how to enable a given report (assuming they know it exists), modify their code, and rerun.
+
+This POEM proposes to turn on common reports **by default**.
+This gives novice users this information for free.
+All reports will be saved within a subfolder to avoid cluttering the current working directory and making cleanup easier.
+Reports will be saved as HTML documents.
+
+A good name for the subfolder might be `f'{sys.argv[0]}_{problem_name}_reports'` in the current working directory, so that multiple run files in the CWD don't overwrite reports made by each other.
+Add `report_dir` as an option to Problem so that the user can override the default.
+
+### Disabling report generation
+
+Experienced users may not want to deal with generating reports on each run of their model, especially if their systems are especially large and the report generation is too time consuming.
+This can be done by using a `reports` argument to Problem.setup (similar to checks), or by setting an environment variable `OPENMDAO_REPORTS`, where a value that is one of `['0', 'false', 'off']` (noisily) disables all report generation.
```

Comment:
why not an option?   alongside the `report_dir` option above...

Reply:
Agreed, an option is probably the right place for this.